src/Microsoft.AspNet.Security/AuthenticationTicket.cs

```diff
@@ -29,11 +23,34 @@ public AuthenticationTicket(ClaimsIdentity identity, AuthenticationProperties pr
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="AuthenticationTicket"/> class
+        /// </summary>
+        /// <param name="identity">the <see cref="ClaimsPrincipal"/> that represents the authenticated user.</param>
+        /// <param name="properties">additional properties that can be consumed by the user or runtims.</param>
+        /// <param name="authenticationType">the authentication middleware that was responsible for this ticket.</param>
+        public AuthenticationTicket(ClaimsPrincipal principal, AuthenticationProperties properties, string authenticationType)
+        {
+            AuthenticationType = authenticationType;
+            Principal = principal;
+            Properties = properties ?? new AuthenticationProperties();
+        }
+
+        /// <summary>
+        /// Gets the authenticated user identity.
+        /// </summary>
+        public string AuthenticationType { get; private set; }
+
+        /// <summary>
         /// Gets the authenticated user identity.
         /// </summary>
         public ClaimsIdentity Identity { get; private set; }
 
         /// <summary>
+        /// Gets the authenticated user identity.
+        /// </summary>
+        public ClaimsPrincipal Principal{ get; private set; }
```

Comment:
FYI: I see you've handled this in InvokeReplyPath, but it's not handled in AuthenticationHandler.BaseInitializeAsync or Authenticate. Because of the flow of this speific OIDC middleware most people won't hit those code paths, but long term that's a good example of why we shouldn't have two different identity properties on the ticket.